u2f-gae-demo/src/com/google/u2f/gaedemo/storage/TokenStorageData.java

```diff
@@ -39,21 +43,24 @@ public TokenStorageData(SecurityKeyData tokenData) {
     } catch (CertificateEncodingException e) {
       throw new RuntimeException();
     }
+    this.transports = tokenData.getTransports();
     this.counter = tokenData.getCounter();
   }
 
   public void updateCounter(int newCounterValue) {
     counter = newCounterValue;
   }
-  
+
   public SecurityKeyData getSecurityKeyData() {
-    return new SecurityKeyData(enrollmentTime, keyHandle, 
-        publicKey, parseCertificate(attestationCert), counter);
+    X509Certificate x509cert = parseCertificate(attestationCert);
+    return new SecurityKeyData(enrollmentTime, transports, keyHandle, 
+        publicKey, x509cert, counter);
   }
   
   public JsonObject toJson() {
     JsonObject json = new JsonObject();
     json.addProperty("enrollment_time", enrollmentTime);
+    json.add("transports", getJsonTransports());
```

Comment:
getJsonTransports() returns an object, not a string. That is so we can iterate over the array of Transports on the client side without having to do string parsing.

addProperty does not take an Object as argument. It only works for strings and numbers (i.e., simple types). For Object types we have to use add.